### PR TITLE
conveyor2hdf

### PR DESCRIPTION
**Pull request by jweissm**

### CODE REVIEW

File: src/commands/convert/conveyor2hdf.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  async run() {
+    const {flags} = await this.parse(Conveyor2HDF)
+
+    // Check for correct input type
+    const data = fs.readFileSync(flags.input, 'utf8')
+    checkInput({data, filename: flags.input}, 'Conveyor', 'Conveyor JSON')
+
+    const converter = new Mapper(data)
+    fs.writeFileSync(checkSuffix(flags.output), JSON.stringify(converter.toHdf()))
```
Comment (Amndeep7) [inline src/commands/convert/conveyor2hdf.ts:27]: if you want to print out the entire result that's fine, but put it in the output directory

File: src/commands/convert/conveyor2hdf.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    fs.mkdirSync(flags.output)
+    _.forOwn(results, (result, filename) => {
```
Comment (Amndeep7) [inline src/commands/convert/conveyor2hdf.ts:29]: you aren't storing the 'results' of doing the toHdf call, so this section of the code will fail

**Review by Amndeep7 — CHANGES_REQUESTED** [review]

File: src/commands/convert/conveyor2hdf.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import {Command, Flags} from '@oclif/core'
```
Comment (Amndeep7) [inline src/commands/convert/conveyor2hdf.ts:1]: https://github.com/mitre/saf/blob/main/src/commands/convert/index.ts
hasn't been added to the generic command

Comment (Amndeep7) [inline src/commands/convert/conveyor2hdf.ts:1]: missing tests and samples

Comment (Amndeep7) [inline src/commands/convert/conveyor2hdf.ts:1]: missing readme documentation

**Review by Amndeep7 — CHANGES_REQUESTED** [review]

File: src/commands/convert/conveyor2hdf.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jweissm) [reply src/commands/convert/conveyor2hdf.ts:29]: fixed

Comment (jweissm) [reply src/commands/convert/conveyor2hdf.ts:29]: fixed

File: src/commands/convert/conveyor2hdf.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    // Check for correct input type
+    const data = fs.readFileSync(flags.input, 'utf8')
+    checkInput({data, filename: flags.input}, 'Conveyor', 'Conveyor JSON')
+
+    const converter = new Mapper(data)
+    fs.writeFileSync(checkSuffix(flags.output), JSON.stringify(converter.toHdf()))
+    fs.mkdirSync(flags.output)
+    _.forOwn(results, (result, filename) => {
+      fs.writeFileSync(
+        path.join(flags.output, checkSuffix(filename)),
```
[the remark this answers is not in the document]
Comment (jweissm) [reply src/commands/convert/conveyor2hdf.ts:31]: fixed

File: src/commands/convert/conveyor2hdf.ts (hunk 3 above)
Comment (jweissm) [reply src/commands/convert/conveyor2hdf.ts:1]: added to index.ts

Comment (jweissm) [reply src/commands/convert/conveyor2hdf.ts:1]: added

Comment (jweissm) [reply src/commands/convert/conveyor2hdf.ts:1]: added samples and test

File: src/commands/convert/conveyor2hdf.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    const {flags} = await this.parse(Conveyor2HDF)
+
+    // Check for correct input type
+    const data = fs.readFileSync(flags.input, 'utf8')
+    checkInput({data, filename: flags.input}, 'Conveyor', 'Conveyor JSON')
+
+    const converter = new Mapper(data)
+    const results = converter.toHdf()
+    fs.mkdirSync(flags.output)
+    for(const [result, filename] of Object.entries(results) ){
```
Comment (Amndeep7) [inline src/commands/convert/conveyor2hdf.ts:29]: please address github linter failures

image

File: src/commands/convert/index.ts [hunk 6]
```diff
@@ -96,6 +97,20 @@ export default class Convert extends Command {
         break
       }
 
+      case 'conveyor': {
```
Comment (Amndeep7) [inline src/commands/convert/index.ts:100]: there's a merge conflict in this file still

**Review by Amndeep7 — CHANGES_REQUESTED** [review]

File: src/commands/convert/conveyor2hdf.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Amndeep7) [reply src/commands/convert/conveyor2hdf.ts:30]: please use object.entries instead of lodash here

**Review by Amndeep7 — APPROVED** [review]

**Pull request merged**